### DISCUSSION TITLE
Belay with STM32/Pyboard from Windows 10 PC

### DISCUSSION THREAD

**Discussion by rwaynecollins** [micropython/micropython#12501] [STM32 / Pyboard]
I am trying to get Belay working in python on Windows 10. I Just discovered "Belay" yesterday and it seems to be able to do what I need, but I can't get it to work. I can't get the initial Device/Port assignment to work.

I have my STM32F405 pyboard connected to a usb port on my Windows 10 PC and it shows up as "USB Serial Device (COM9)". My problem is I cannot get the device assignment to work without errors. I think the problem is I am not specifying the port properly.

Examples show the assignment as follows
device = belay.Device("/dev/ttyUSB0")
but it does not work, I get "ConnectionFailedError"

I have tried different variations of the port specification lile "/dev/ttyUSB9", "COM9", etc. but nothing works.
Any advice on how to get this connection to work in Windows 10 using python and "micropython on the pyboard" ?

Example program I am working with:
------------------------
# -*- coding: utf-8 -*-
"""
Created on Thu Sep 21 16:00:09 2023

@author: wcollins
"""

import belay

device = belay.Device("/dev/ttyUSB0")
print("Device Assigned....")

@device.task
def set_led(state):
    print(f"Printing from device; turning LED to {state}.")
    Pin(25, Pin.OUT).value(state)
    
       
set_led(True)
--------------------------------
Thanks in advance
Wayne

**Comment by rkompass** [comment]
You may try `python -m serial.tools.list_ports` in a terminal with python installed and your pyboard plugged in.
On Linux I get `/dev/ttyACM0` but on Windows there should also be a port displayed.

**Reply by rwaynecollins** [reply]
It says the port is COM15
Tried  device = belay.Device("COM15")
I get this
Connecting Device...
Traceback (most recent call last):
  File "belay_test2.py", line 12, in <module>
    device = belay.Device("COM15")
  File "C:\Users\wcollins\AppData\Local\Programs\Spyder\Python\lib\site-packages\belay\device.py", line 107, in __init__
    self._exec_snippet("startup")
  File "C:\Users\wcollins\AppData\Local\Programs\Spyder\Python\lib\site-packages\belay\device.py", line 226, in _exec_snippet
    snippets = [read_snippet(name) for name in names]
  File "C:\Users\wcollins\AppData\Local\Programs\Spyder\Python\lib\site-packages\belay\device.py", line 226, in <listcomp>
    snippets = [read_snippet(name) for name in names]
  File "C:\Users\wcollins\AppData\Local\Programs\Spyder\Python\lib\site-packages\belay\helpers.py", line 23, in read_snippet
    return importlib.resources.files(snippets).joinpath(resource).read_text()
AttributeError: module 'importlib.resources' has no attribute 'files'

**Reply by rkompass** [reply]
This is an issue you might report on the Belay GitHub site.
Also you might consider trying it with WSL or a virtual Linux.

My impression is that most of those following the discussions here have little or no experience with Belay.
I tried it once without problem (under Linux) and had the impression, it would be a nice tool to integrate the control of several
MP devices interacting with each other from one command line / python program.
However now it seems this feature shall be integrated into mpremote.
